### ISSUE TITLE
feat: add support to register into venom.schema.json user defined executors

### ISSUE THREAD

**Issue by ivan-velasco** [Thiht/vscode-venom#6]
```yaml
testcases:
  - name: get_login_token
    steps:
    - type: get_login_token
```

Since this type is not present in the `venom.schema.json` we get an error in the editor

user defined executors is a common functionality used to avoid duplicate code. To have the plugin register the defined types in `lib` directory defined in the `.venomrc` would be really useful

Definition of the user defined executor:
```yaml
executor: get_login_token
input:
  arg: {}
steps:
  - script: sh ../../resources/scripts/gettoken.sh
    
output: 
   token: "{{.result.systemout}}"
```

**Comment by Thiht** [comment]
Hi,
Thanks for the suggestion! I never used user-defined executor so I'll see if I can generate a JSON schema directly from the YAML declarations. Seems doable at first glance.

**Comment by Thiht** [comment]
Sorry it took so long, it was harder than expected :D
The change will be released in 1.7.0.

Custom executors are expected to be in `lib/` by default, but it can be overridden in the `.venomrc` via the `lib_dir` setting.

Custom executors are added dynamically to the JSON schema of test suites, so you get auto complete of the custom steps and their `inputs`:

<img width="235" alt="Capture d’écran 2023-12-01 à 15 17 35" src="https://github.com/Thiht/vscode-venom/assets/2796344/d61d313d-121c-496b-b47c-e8808e4195d7">

<img width="235" alt="Capture d’écran 2023-12-01 à 15 18 00" src="https://github.com/Thiht/vscode-venom/assets/2796344/fb4caf03-0b0d-450d-9211-6f5efbdb81f0">

There are a few things I'll improve in the future:
- the JSON schema of custom executors themselves is not full featured yet, specifically when you write a custom executor there's no autocomplete on `step` types and value like in test suites
- if you update the `lib_dir` in a `.venomrc` I think you'll have to reload VSCode for it to be taken in consideration
